### PR TITLE
update rug to 0.9, and use rug more idiomatically

### PR DESCRIPTION
**Pull request by tspiteri**
This commit:

* updates the rug dependency to the latest version
* only uses the required features of rug to save rug compilation time as it doesn't have to compile the float and complex libraries
* uses rug more idiomatically

### CODE REVIEW

**Comment by jonhoo** [comment]
I have half a mind to leave the `rug` dependency on with default features, just because then users of the crate who also use `rug` won't have to compile it twice. But not sure what the right policy is there. Otherwise it looks good barring the inline question for @marshallpierce

File: tests/quantile.rs [hunk 1]
```diff
@@ -373,16 +369,8 @@ impl<'a, R: Rng + 'a> Iterator for RandomMaxIter<'a, R> {
...
-
-    // emulate ceil()
-    if product == Rational::from(prod_as_int.clone()) {
-        return prod_as_int.to_u64().unwrap();
-    } else {
-        // to_integer()'s rounding down has chopped off a fractional part
-        return prod_as_int.to_u64().unwrap() + 1;
-    }
+    let product = Rational::from_f64(quantile).unwrap() * Rational::from(count);
+    product.ceil().to_u64().unwrap()
```
Comment (jonhoo) [inline tests/quantile.rs]: @marshallpierce I know you've had a fair bit of time to play with these corner cases. Do you think rug's `ceil` does what you expect?

**Comment by tspiteri** [comment]
The `rug` crate is only a dev-dependency in this case, so users of the `hdrsample` crate do not have to compile `rug` at all unless they are using it directly. Besides, if

* crate `r` has features `rat` and `flo`,
* crate `h` depends on `r` using only `rat`, and
* crate `u` depends on `h`, and depends on `r` directly using only `flo`, then

crate `r` will be compiled once with both features `rat` and `flo` enabled.

File: tests/quantile.rs (hunk 1 above)
Comment (marshallpierce) [reply tests/quantile.rs]: It looks fine; I can't recall why I avoided using it to begin with.

File: Cargo.toml [hunk 2]
```diff
@@ -36,7 +36,7 @@ base64 = { version = "0.7.0", optional = true }
 
 [dev-dependencies]
 rand = "0.3.15"
-rug = "0.6.0"
+rug = { version = "0.9.0", default-features = false, features = ["rational"] }
```
Comment (marshallpierce) [inline Cargo.toml]: Looks like 0.9.1 is the new hotness.

**Comment by marshallpierce** [comment]
I'm also unclear on why we should bother with changing the features enabled for `rug` -- only people who are running `hdrsample`'s tests would need to compile it, right?

**Comment by tspiteri** [comment]
Yes, it's people who are running the tests than need to compile it, I thought saving twenty seconds there couldn't hurt.

File: Cargo.toml (hunk 2 above)
Comment (tspiteri) [reply Cargo.toml]: I have a habit of using the lowest version that is semver-compatible with the latest version and that provides all the required functionality.(0.9.0 is semver-compatible with 0.9.1, so cargo will get 0.9.1 anyway, and it has all the used functionality.)

Comment (marshallpierce) [reply Cargo.toml]: If cargo will resolve it to 0.9.1, let's specify 0.9.1. I view the default cargo behavior, when combined with the default advice of not checking in Cargo.lock, as a terrible mis-feature. I forget all the time that I have to add a `^` to make Cargo do what I want or I would have had it `^`'d from the beginning. :(

**Comment by marshallpierce** [comment]
OK. I don't really care one way or another; one-time compile costs don't really bother me (as opposed to compilation costs that I have to pay every time I run the tests) but I don't mind removing them if it bothers other folks.

File: Cargo.toml (hunk 2 above)
Comment (tspiteri) [reply Cargo.toml]: I'll change it to 0.9.1 then, and I'll also remove the features stuff. As you said below, one time costs during testing are not a real issue, so the feature stuff is basically noise in the Cargo.toml file.

**Comment by jonhoo** [comment]
Thanks @tspiteri!
